docs/source/tutorials/06_using_your_own_models.rst

```diff
@@ -0,0 +1,578 @@
+
+   ``casting_classifier.yml`` updated content:
+
+   .. code-block:: yaml
+      :linenos:
+
+      input: ["img"]
+      output: ["pred_label", "pred_score"]
+
+      weights_parent_dir: weights
+      class_label_map: {0: "defective", 1: "normal"}
+
+#. **src/custom_nodes/model/casting_classifier.py**:
+
+   ``casting_classifier.py`` updated content:
+
+   .. container:: toggle
+
+      .. container:: header
+
+         **Show/Hide Code for casting_classifier.py**
+
+      .. code-block:: python
+         :linenos:
+
+         """
+         Casting classification model.
+         """
+
+         from typing import Any, Dict
+
+         import cv2
+         import numpy as np
+         import tensorflow as tf
+
+         from peekingduck.pipeline.nodes.node import AbstractNode
+
+         IMG_HEIGHT = 180
+         IMG_WIDTH = 180
+
+
+         class Node(AbstractNode):
+            """Initializes and uses a CNN to predict if an image frame shows a normal
+            or defective casting.
+            """
+
+            def __init__(self, config: Dict[str, Any] = None, **kwargs: Any) -> None:
+               super().__init__(config, node_path=__name__, **kwargs)
+               self.model = tf.keras.models.load_model(self.weights_parent_dir)
+
+            def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+               """Reads the image input and returns the predicted class label and
+               confidence score.
+
+               Args:
+                     inputs (dict): Dictionary with key "img".
+
+               Returns:
+                     outputs (dict): Dictionary with keys "pred_label" and "pred_score".
+               """
+               img = cv2.cvtColor(inputs["img"], cv2.COLOR_BGR2RGB)
+               img = cv2.resize(img, (IMG_WIDTH, IMG_HEIGHT))
+               img = np.expand_dims(img, axis=0)
+               predictions = self.model.predict(img)
+               score = tf.nn.softmax(predictions[0])
+
+               return {
+                     "pred_label": self.class_label_map[np.argmax(score)],
+                     "pred_score": 100.0 * np.max(score),
+               }
+
+The custom node takes in the in-built PeekingDuck :term:`img` data type, makes a prediction based
+on the image, and produces two custom data types: `pred_label`, the predicted label ("defective"
+or "normal"); and `pred_score`, which is the confidence score of the prediction.
+
+
+Using the Classifier in a PeekingDuck Pipeline
+----------------------------------------------
+
+We'll now pair this custom node with other PeekingDuck nodes to build a complete solution. Imagine
+an automated inspection system like the one shown below, where the castings are placed on a
+conveyor belt and a camera takes a picture of each casting and sends it to the PeekingDuck pipeline
+for prediction. A report showing the predicted result for each casting is produced for the quality
+inspector, and he/she can use the report for further analysis. 
```

Comment:
Suggest rephrasing the above as: `A report showing the predicted result for each casting is produced, and the quality inspector can use it for further analysis.`

Reply:
This sounds better, rephrased.